packages/flutter/lib/src/painting/circle_border.dart

```diff
@@ -80,37 +79,25 @@ class CircleBorder extends OutlinedBorder {
 
   @override
   Path getInnerPath(Rect rect, { TextDirection? textDirection }) {
-    final double radius = rect.shortestSide / 2.0;
-    final double adjustedRadius;
+    final double delta;
     switch (side.strokeAlign) {
       case StrokeAlign.inside:
-        adjustedRadius = radius - side.width;
+        delta = side.width;
         break;
       case StrokeAlign.center:
-        adjustedRadius = radius - side.width / 2.0;
+        delta = side.width / 2.0;
         break;
       case StrokeAlign.outside:
-        adjustedRadius = radius;
+        delta = 0;
         break;
     }
-    return Path()
-      ..addOval(Rect.fromCircle(
-        center: rect.center,
-        radius: math.max(0.0, adjustedRadius),
-      ));
+    final Rect adjustedRect = _adjustRect(rect).deflate(delta);
+    return Path()..addOval(adjustedRect);
   }
 
   @override
   Path getOuterPath(Rect rect, { TextDirection? textDirection }) {
-    if (ovalness != 0) {
       return Path()..addOval(_adjustRect(rect));
-    }
-
-    return Path()
-      ..addOval(Rect.fromCircle(
-        center: rect.center,
-        radius: rect.shortestSide / 2.0,
-      ));
```

Comment:
Why did you remove it?

Reply:
Because I made it so the _adjustRect() can return value because it is already using that condition internally!!